language/move-prover/tests/sources/functional/verify_vector.move

```diff
@@ -97,22 +97,43 @@ module VerifyVector {
         ensures result[0] == e;
     }
 
+    spec module {
+        /// Ghost variable `old_v` used to store the old value of the array v.
+        /// TODO: by allowing old(.) to appear in assume/assert expressions,
+        /// we can eliminate the need for old_v.
+        global old_v<Element>: vector<Element>;
+    }
+
     // Reverses the order of the elements in the vector in place.
     fun verify_reverse<Element>(v: &mut vector<Element>) {
-        let len = Vector::length(v);
-        if (len == 0) return ();
+        let vlen = Vector::length(v);
+        if (vlen == 0) return ();
+
+        spec {
+            /// Initialize the old vector `old_v` ghost variable
+            assume forall k in 0..vlen: old_v<Element>[k] == v[k];
```

Comment:
Can probably replace it with `assume old_v<Element> == v`, although not important if we implement some version of old(.).